source/dub/dub.d

```diff
@@ -417,6 +417,14 @@ class Dub {
 	@property const(string[string]) defaultPreRunEnvironments() const { return this.m_config.defaultPreRunEnvironments; }
 	@property const(string[string]) defaultPostRunEnvironments() const { return this.m_config.defaultPostRunEnvironments; }
 
+	/** An optional directory that is used for storing cached build artefacts.
+
+		This is used by `BuildGenerator` to store intermediate object files
+		for reuse in incremental builds. The value of this proprerty may be
+		an empty path, in which case no directory is configured.
+	*/
+	@property NativePath buildCacheDirectory() const { return m_config.buildCacheDirectory; }
+
```

Comment:
Instead of adding piece-meal getters, which I think doesn't scale, we could just return the `m_config`?

Reply:
Hm, not sure about that. `UserConfiguration` (unless I'm mistaken) represents the format of the settings file. Tying that to the public API and at the same time considering changing the file format makes it sound like a leaky abstraction that we should probably avoid.

Reply:
It is indeed. But I don't see why we are exposing `buildCacheDirectory` and other settings. It seems that every time someone touches the configuration, they just cargo-cult-ishly copy the existing approach.